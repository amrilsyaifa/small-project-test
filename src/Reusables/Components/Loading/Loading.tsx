import { Spin } from 'antd';
import React from 'react';
import Styles from './Loading.module.scss';

const Loading: React.FC = () => (
  <div className={Styles['wrapper']}>
    <Spin tip="Loading..." />
  </div>
);

export default Loading;
