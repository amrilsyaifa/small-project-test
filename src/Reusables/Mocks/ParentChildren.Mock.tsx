import React from 'react';

/**
 * Separate children from Component so it's not circular referencing when stringified
 */
interface IParentChildrenSeparator {
  mockLabel: string;
  children: React.ReactNode;
  MockElement?: React.FC;
}
export const ParentChildrenSeparator: React.FC<IParentChildrenSeparator> = ({
  mockLabel,
  children,
  MockElement,
  ...props
}) => {
  const RenderElement = MockElement ? MockElement : MockDivElement;

  return (
    <RenderElement data-testid={mockLabel} data-props={JSON.stringify(props)}>
      {children}
    </RenderElement>
  );
};

export const MockDivElement = ({ ...props }) => <div {...props} />;

export default ParentChildrenSeparator;
