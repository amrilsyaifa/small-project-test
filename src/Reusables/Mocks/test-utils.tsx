import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { persistedReducer } from 'src/Store';
// Import your own reducer

interface IWrapper {
  children: React.ReactNode;
}

const render = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  { store = configureStore({ reducer: persistedReducer }), ...renderOptions } = {}
) => {
  const Wrapper: React.FC<IWrapper> = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
