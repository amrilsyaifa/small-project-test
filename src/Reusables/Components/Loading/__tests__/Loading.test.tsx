import { render, cleanup } from 'src/Reusables/Mocks/test-utils';
import React from 'react';
import Loading from '../Loading';
import { MatchMediaObject } from 'src/Reusables/Mocks/MatchMediaObjectTest';

afterEach(cleanup);
beforeEach(() => {
  MatchMediaObject();
});

describe('Loading', () => {
  test('Loading Render Snapshoot', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
