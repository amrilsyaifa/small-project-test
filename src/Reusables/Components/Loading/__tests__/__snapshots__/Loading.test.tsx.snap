// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Loading Render Snapshoot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-spinning ant-spin-show-text"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
      <div
        class="ant-spin-text"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
